openlibrary/core/models.py

```diff
@@ -827,6 +832,33 @@ def get_loans_per_day(self, resource_type="total"):
         stats = loanstats.LoanStats(library=name)
         return stats.get_loans_per_day(resource_type=resource_type)
 
+class UserGroup(web.storage):
+
+    @classmethod
+    def add_user(cls, userkey, usergroup):
+        """Administrative utility (designed to be used in conjunction with
+        accounts.escalate_privilege_and_run_as) to add a patron to a usergroup
+
+        :param str userkey: e.g. /people/mekBot
+        :param str usergroup: e.g. /usergroup/sponsor-waitlist
+        :rtype: str
+        :return: the string "success" or an error string
+        """
+        # make sure user_key exists
+        if web.ctx.site.get(userkey):
+            doc = web.ctx.site.get("/usergroup/%s" % usergroup)
+            if doc:
+                group = doc.dict()
+                # Make sure userkey not already in group members:
+                group.setdefault('members', [])
+                if not any(userkey == member['key'] for member in group['members']):
```

Comment:
If we make `UserGroup` inherit from `Thing`, then we can put this in a helper method:

```py
self.has_member(userkey)
```